### PR TITLE
add plugin integration tests

### PR DESCRIPTION
**Pull request by antipeon**

### CODE REVIEW

File: docker/action-scripts/build-and-unit-tests.sh [hunk 1]
```diff
@@ -7,6 +7,6 @@
 source docker/building_dependencies/runtime_env.sh
 chmod +x docker/action-scripts/build-utbot.sh
 ./docker/action-scripts/build-utbot.sh
-cd server/build
-chmod +x UTBot_UnitTests
-./UTBot_UnitTests --verbosity info --log `pwd`
+#cd server/build
+#chmod +x UTBot_UnitTests
+#./UTBot_UnitTests --verbosity info --log `pwd`
```
Comment (antipeon) [inline docker/action-scripts/build-and-unit-tests.sh:12]: need restore

File: .github/workflows/build-utbot.yml [hunk 2]
```diff
@@ -46,12 +46,21 @@ jobs:
...
+          ./docker/action-scripts/build-vsix.sh
+#      - name: Install Node.js
+#        uses: actions/setup-node@v1
+#        with:
+#          node-version: 15.x
+#      - run: npm install
+#      - name: Run tests
+#        uses: GabrielBB/xvfb-action@v1.0
+#        with:
+#          run: npm test ./integration-tests/c-example --extensionDevelopmentPath=./vscode-plugin --extensionTestsPath=./vscode-plugin/out/test/suite/index
```
Comment (antipeon) [inline .github/workflows/build-utbot.yml:61]: remove

File: docker/action-scripts/build-vsix-and-integration-tests.sh [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+npm version $VERSION --allow-same-version
+utbot_dir="$(dirname $PWD)"
+npm install -g vsce --unsafe
+npm install --unsafe
+npm rebuild grpc --runtime=electron --target=7.3.0
+chmod +x protoc.sh && ./protoc.sh $utbot_dir/server/proto $utbot_dir/vscode-plugin/src/proto-ts
+npm run compile
+vsce package --no-yarn
+
+#run integration tests
```
Comment (ladisgin) [inline docker/action-scripts/build-vsix-and-integration-tests.sh:22]: empty scope

File: integration-tests/c-example-mini/.gitignore [hunk 4]
```diff
@@ -0,0 +1,4 @@
+.vscode/
+build/
+tests/
+utbot-tests/
```
Comment (ladisgin) [inline integration-tests/c-example-mini/.gitignore:4]: add endline

**Review by ladisgin — COMMENTED** [review]
Probably we can use separate CMakeLists.txt or some scripts instead of copy-paste c-example files?

File: .github/workflows/build-utbot.yml [hunk 5]
```diff
@@ -41,17 +41,53 @@ jobs:
...
+          submodules: recursive
+
+      - name: build UTBot
+        run: |
+          chmod +x docker/action-scripts/build-utbot.sh
+          ./docker/action-scripts/build-utbot.sh
+
+      - name: Run Integration Tests
         run: |
+          chmod +x server/build/utbot
```
Comment (ladisgin) [inline .github/workflows/build-utbot.yml:89]: useless

File: docker/action-scripts/integration-tests.sh [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+utbot_dir="$(dirname $PWD)"
+
+
+#download vs_code
+npm run download_vscode
+
+set -e
+#Path variables
+VSCODE_VERSION_DIR=$(ls -d -1 $utbot_dir/vscode-plugin/.vscode-test/vscode-* |sed -n '1p')
+echo "VSCODE_VERSION_DIR=$VSCODE_VERSION_DIR"
```
Comment (ladisgin) [inline docker/action-scripts/integration-tests.sh:18]: duplicate vscode-plugin/src/test/run.sh

File: vscode-plugin/src/test/suite/projectTests.test.ts [hunk 7]
```diff
@@ -16,31 +16,22 @@ import {
         checkTestFileNotEmpty,
         checkTestFilesGenerated,
         Compiler,
-        openFile,
+        openFile, PARAMETRIZED_TEST_MODE,
```
Comment (ladisgin) [inline vscode-plugin/src/test/suite/projectTests.test.ts:19]: one var per line

**Review by ladisgin — CHANGES_REQUESTED** [review]

**Review by ladisgin — APPROVED** [review]

**Pull request merged**